### PR TITLE
storagenode/pieces: allow to configure initial piece scan

### PR DESCRIPTION
**Pull request by odarriba**
What: 

Allow SNOs to configure if they want the initial piece scan that is executed during the start up of the node

Why:

Because for SNOs that run several nodes in different disks it requires a lot of IO when the nodes become big enough and the disk is starting to be fragmented.

It is not a feature which shoulds be disabled all the time, but it can allow the SNOs tio decide when they want to execute it, which is something desirable now that the node is updated manually if you use the official Docker image.

It would be nice to document also this function either in the docs site or in the forum, as disabling this initial scan and having a lot of dirty restarts of the storagenode can lead to mismatches between the stored disk usage and the real one.

More information about this config flag can be found [on this forum thread](https://forum.storj.io/t/tuning-the-filewalker/19203)

Please describe the tests:

No tests have been added for this feature, as it is a simple configuration flag. I'm not an experiences Golang developer, but if you feel it is mandatory for this kind of feature I can try to add them.
 
Please describe the performance impact:

It has a huge impact on IO performance for big nodes on restarts of the storagenode process.

Also, the impact on metric should not be big if the storagenode is shutted down gracefully, as the current usage is stored also on a database.

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: storagenode/piecestore/endpoint.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type Config struct {
 	DeleteQueueSize         int           `help:"size of the piece delete queue" default:"10000"`
 	OrderLimitGracePeriod   time.Duration `help:"how long after OrderLimit creation date are OrderLimits no longer accepted" default:"1h0m0s"`
 	CacheSyncInterval       time.Duration `help:"how often the space used cache is synced to persistent storage" releaseDefault:"1h0m0s" devDefault:"0h1m0s"`
+	InitialCheck            bool          `help:"if all pieces usage should be recalculated on startup" releaseDefault:"true" devDefault:"true"`
```
Comment (littleskunk) [inline storagenode/piecestore/endpoint.go:63]: If dev and release default are the same please only specify default. See the config line below for an example.

File: storagenode/piecestore/endpoint.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type Config struct {
 	DeleteQueueSize         int           `help:"size of the piece delete queue" default:"10000"`
 	OrderLimitGracePeriod   time.Duration `help:"how long after OrderLimit creation date are OrderLimits no longer accepted" default:"1h0m0s"`
 	CacheSyncInterval       time.Duration `help:"how often the space used cache is synced to persistent storage" releaseDefault:"1h0m0s" devDefault:"0h1m0s"`
+	PieceScanOnStart				bool          `help:"ifset to true, all pieces disk usage is recalculated on startup" default:"true"`
```
Comment (profclems) [inline storagenode/piecestore/endpoint.go:63]: Seems it’s not gofmt’ed. You could run `make goimports-fix` to properly format it.

**Review by profclems — CHANGES_REQUESTED** [review]

File: storagenode/piecestore/endpoint.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ type Config struct {
 	DeleteQueueSize         int           `help:"size of the piece delete queue" default:"10000"`
 	OrderLimitGracePeriod   time.Duration `help:"how long after OrderLimit creation date are OrderLimits no longer accepted" default:"1h0m0s"`
 	CacheSyncInterval       time.Duration `help:"how often the space used cache is synced to persistent storage" releaseDefault:"1h0m0s" devDefault:"0h1m0s"`
-	InitialCheck            bool          `help:"if all pieces usage should be recalculated on startup" default:"true"`
+	PieceScanOnStart				bool          `help:"ifset to true, all pieces disk usage is recalculated on startup" default:"true"`
```
Comment (profclems) [inline storagenode/piecestore/endpoint.go:63]: Should we use `PieceScanOnStartup`?

File: storagenode/pieces/cache.go [hunk 4]
```diff
@@ -33,12 +34,13 @@ type CacheService struct {
 
 // NewService creates a new cache service that updates the space usage cache on startup and syncs the cache values to
 // persistent storage on an interval.
-func NewService(log *zap.Logger, usageCache *BlobsUsageCache, pieces *Store, interval time.Duration) *CacheService {
+func NewService(log *zap.Logger, usageCache *BlobsUsageCache, pieces *Store, interval time.Duration, PieceScanOnStartup bool) *CacheService {
```
Comment (profclems) [inline storagenode/pieces/cache.go:37]: Still failing lint: `PieceScanOnStartup` should not be capitalised as a function parameter

Comment (odarriba) [reply storagenode/pieces/cache.go]: Fixed! Sorry for that.

The CI keeps failing but I’m not able to know which test fails 😅

**Review by thepaul — APPROVED** [review]
I love it

**Comment by thepaul** [comment]
It looks like the build failures are unrelated to this PR, so don't worry about that.

File: storagenode/pieces/cache.go (hunk 4 above)
Comment (profclems) [reply storagenode/pieces/cache.go]: Yeah the build failures are not related to your change. We'll have a look

File: storagenode/piecestore/endpoint.go [hunk 5]
```diff
@@ -60,6 +60,7 @@ type Config struct {
 	DeleteQueueSize         int           `help:"size of the piece delete queue" default:"10000"`
 	OrderLimitGracePeriod   time.Duration `help:"how long after OrderLimit creation date are OrderLimits no longer accepted" default:"1h0m0s"`
 	CacheSyncInterval       time.Duration `help:"how often the space used cache is synced to persistent storage" releaseDefault:"1h0m0s" devDefault:"0h1m0s"`
+	PieceScanOnStartup      bool          `help:"ifset to true, all pieces disk usage is recalculated on startup" default:"true"`
```
[the remark this answers is not in the document]
Comment (odarriba) [reply storagenode/piecestore/endpoint.go]: That's tru! Fixed in https://github.com/storj/storj/pull/5024/commits/e2b2842102bd54d250af65db41aa0029023889de

File: storagenode/pieces/cache.go (hunk 4 above)
Comment (odarriba) [reply storagenode/pieces/cache.go]: Fixed in https://github.com/storj/storj/pull/5024/commits/4908eda1137fb8547c4cf52b925f208dff9cf19f !

**Review by thepaul — APPROVED** [review]

**Review by profclems — APPROVED** [review]

**Pull request merged**